### PR TITLE
Add and wire up swagger file for admin API

### PR DESCRIPTION
**Pull request by reggieriser**
## Description

This PR adds and wires up a swagger file for the admin API.

## Reviewer Notes

Note that I have a placeholder endpoint with no implementation currently in the `admin.yaml` swagger file.  Without any endpoints, go-swagger will throw an error.  This will be removed as we add actual admin endpoints.

## Setup

1. `make server_run`
2. `make admin_client_run`
3. Try the following local URLs:
  - http://adminlocal:3000/admin/v1/docs -- the swagger UI
  - http://adminlocal:3000/admin/v1/swagger.yaml -- the swagger file

There's also a login page in place for admin right now like our other apps, found at http://adminlocal:3000/ .  This login process may change some once we wire in auth for React Admin, but having the login allows us to test login.gov integration and other wiring that goes along with that.  Note that in development right now we're getting back different UUIDs (which we store in the `users` table) from login.gov for each of our apps, so you will have to do some database manipulation to test a successful login:

1. Try to login to admin.  You should get an `Unauthorized` response after authentication.  This is because there's no `users` record with the UUID that came back from login.gov.
2. In `authentication/auth.go`, either set a breakpoint on the call to `FetchUserIdentity` and examine the UserID UUID there, or turn on pop debugging around that call to see the SQL locally.  Note the UUID that it's looking for.
3. Create a `users` record with `login_gov_uuid` set to the UUID noted in step 2.  Use your `login.gov` email for the `login_gov_email` column.  Make sure you set `is_superuser` to true.  The `id` for the record can be any UUID you generate.
4. Try logging in again -- you should get a page saying `React admin is properly configured`.

We plan to address the multiple UUIDs from login.gov in a later story -- it may affect more than just the admin app. 
## Code Review Verification Steps

* [ ] Code follows the guidelines for [Logging](./docs/backend.md#logging)
* [ ] This works in [Supported Browsers](./docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/164911631) for this change
* [Blocking infra story](https://www.pivotaltracker.com/story/show/165153305)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -165,6 +165,7 @@ commands:
             LOGIN_GOV_MY_CLIENT_ID: urn:gov:gsa:openidconnect.profiles:sp:sso:dod:mymovemillocal
             LOGIN_GOV_OFFICE_CLIENT_ID: urn:gov:gsa:openidconnect.profiles:sp:sso:dod:officemovemillocal
             LOGIN_GOV_TSP_CLIENT_ID: urn:gov:gsa:openidconnect.profiles:sp:sso:dod:tspmovemillocal
+            LOGIN_GOV_ADMIN_CLIENT_ID: urn:gov:gsa:openidconnect.profiles:sp:sso:dod:adminmovemillocal
```
Comment (chrisgilmerproj) [inline .circleci/config.yml]: I still don't have this made. I'll try to figure out how to make the cert shortly.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -80,6 +80,13 @@ repos:
       entry: bin/swagger validate swagger/dps.yaml
       language: script
 
+  - repo: local
+    hooks:
+      - id: swagger
+        name: Admin Swagger
+        entry: bin/swagger validate swagger/admin.yaml
+        language: script
```
Comment (chrisgilmerproj) [inline .pre-commit-config.yaml]: Should these have `passfilenames: false` in them?

File: swagger/admin.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  /placeholder:
+    get:
+      summary: Placeholder until we get real endpoints
+      description: Just a placeholder
+      operationId: placeholder
+      tags:
+        - placeholder
+      responses:
+        200:
+          description: nothing to see here
```
Comment (chrisgilmerproj) [inline swagger/admin.yaml]: I think this would be a 501 - not implemented :-P

File: .pre-commit-config.yaml (hunk 2 above)
Comment (reggieriser) [reply .pre-commit-config.yaml]: I just changed all 5 swagger pre-commit hooks to have a `files` key for the specific swagger YAML file being tested and removed the filename from being explicit on the `entry`.  Now the pre-commit hook for a swagger file should only run if the file has changed.

File: swagger/admin.yaml (hunk 3 above)
Comment (reggieriser) [reply swagger/admin.yaml]: Fixed.

File: .circleci/config.yml (hunk 1 above)
Comment (chrisgilmerproj) [reply .circleci/config.yml]: Done, I have created the app.  You are now set to use this so try it out!

Comment (reggieriser) [reply .circleci/config.yml]: @chrisgilmerproj, I've set up a test login page for adminlocal and am getting back a UUID from login.gov, so things seem to work as far as the provider cert goes.  There's the other issue about different UUIDs coming back for different domains that we're still discussing with the login.gov folks, but I'd like to make a different story to tackle that if it's OK with you so I can get this PR in and unblock some other stories.  Is there anything else you think we need in this PR?

I believe you also wanted to add the other apps after we tested adminlocal.

Comment (chrisgilmerproj) [reply .circleci/config.yml]: Turns out we can re-use the certs already in the dashboard. We only have one cert per environment.  And I'm ok with you tackling the UUID problem in another PR.

**Comment by LeDeep** [comment]
This looks good. I had a branch that did the routing slightly differently. I didn't merge because I didn't want to cause more conflicts with this one, but I might revisit. I'm thinking this works just fine and maybe my potential refactor would be superfluous. I'll run it by you if I think it might be worthwhile to update. Shouldn't hold up this PR, though. Ship it!

**Pull request merged**